From bda97e4f914e18d9d7b2de9b0e9bdc4014c73855 Mon Sep 17 00:00:00 2001
From: Mark Reynolds <mreynolds@redhat.com>
Date: Mon, 25 Nov 2013 09:36:25 -0500
Subject: [PATCH 59/65] Ticket 47599 - fix memory leak

Coverity 12410

https://fedorahosted.org/389/ticket/47599

Reviewed by: richm(Thanks!)
(cherry picked from commit a16bf1b3c4ff0412c2481baace9b427750c11f8c)
(cherry picked from commit c7e7c68ea4779601ec4896b1d90f3b6d347f047f)
(cherry picked from commit 08dc37dc832e1ce78d27012a60b1691dba2f6501)
---
 ldap/servers/slapd/back-ldbm/seq.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ldap/servers/slapd/back-ldbm/seq.c b/ldap/servers/slapd/back-ldbm/seq.c
index 27da2a4..10484fd 100644
--- a/ldap/servers/slapd/back-ldbm/seq.c
+++ b/ldap/servers/slapd/back-ldbm/seq.c
@@ -242,6 +242,7 @@ ldbm_back_seq( Slapi_PBlock *pb )
 				key.flags = 0;
 				for (retry_count = 0; retry_count < IDL_FETCH_RETRY_COUNT; retry_count++) {
 					err = NEW_IDL_DEFAULT;
+					idl_free(idl);
 					idl = idl_fetch( be, db, &key, txn.back_txn_txn, ai, &err );
 					if(err == DB_LOCK_DEADLOCK) {
 						ldbm_nasty("ldbm_back_seq deadlock retry", 1600, err);
-- 
1.8.1.4

